atlasdb-client/src/test/java/com/palantir/atlasdb/util/MetricsRule.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.util;
+
+import org.junit.rules.ExternalResource;
+
+import com.codahale.metrics.ConsoleReporter;
+import com.codahale.metrics.MetricRegistry;
+
+public class MetricsRule extends ExternalResource {
+
+    private MetricRegistry previousMetrics;
+
+    @Override
+    protected void before() throws Throwable {
```

Comment:
Maybe a stupid question, but I am very confused as ``previousMetrics`` and ``metrics`` will be the same thing unless someone else ``sets``(only one call to set is made, which is in this class), noone else does and why are we reporting from the ``newMetrics`` after we reset the registry on line 37.